nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureBlobStorage.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    @Override
+    protected String getPath(final ProcessContext context) {
+        return context.getProperty(AzureConstants.CONTAINER).evaluateAttributeExpressions().getValue();
+    }
+
+    @Override
+    protected boolean isListingResetNecessary(final PropertyDescriptor property) {
+        // re-list if configuration changed, but not when security keys are rolled (not included in the condition)
+        return PREFIX.equals(property)
+                   || AzureConstants.ACCOUNT_NAME.equals(property)
+                   || AzureConstants.CONTAINER.equals(property);
+    }
+
+    @Override
+    protected Scope getStateScope(final ProcessContext context) {
+        return Scope.CLUSTER;
+    }
+
+    @Override
+    protected List<BlobInfo> performListing(final ProcessContext context, final Long minTimestamp) throws IOException {
+        String containerName = context.getProperty(AzureConstants.CONTAINER).evaluateAttributeExpressions().getValue();
+        String prefix = context.getProperty(PREFIX).evaluateAttributeExpressions().getValue();
+        if (prefix == null) {
+            prefix = "";
+        }
+        final List<BlobInfo> listing = new ArrayList<>();
+        try {
+            CloudStorageAccount storageAccount = createStorageConnection(context);
+            CloudBlobClient blobClient = storageAccount.createCloudBlobClient();
+            CloudBlobContainer container = blobClient.getContainerReference(containerName);
+
+            BlobRequestOptions blobRequestOptions = null;
+            OperationContext operationContext = null;
+
+            for (ListBlobItem blob : container.listBlobs(prefix, true, EnumSet.of(BlobListingDetails.METADATA), blobRequestOptions, operationContext)) {
+                if (blob instanceof CloudBlob) {
+                    CloudBlob cloudBlob = (CloudBlob) blob;
+                    BlobProperties properties = cloudBlob.getProperties();
+                    StorageUri uri = cloudBlob.getSnapshotQualifiedStorageUri();
+
+                    Builder builder = new BlobInfo.Builder().primaryUri(uri.getPrimaryUri().toString()).secondaryUri(uri.getSecondaryUri().toString()).contentType(properties.getContentType())
+                            .contentLanguage(properties.getContentLanguage()).etag(properties.getEtag()).lastModifiedTime(properties.getLastModified().getTime()).length(properties.getLength());
+
+                    if (blob instanceof CloudBlockBlob) {
+                        builder.blobType(AzureConstants.BLOCK);
+                    } else {
+                        builder.blobType(AzureConstants.PAGE);
+                    }
+                    listing.add(builder.build());
+                }
+            }
+        } catch (IllegalArgumentException | URISyntaxException | StorageException e) {
+            throw (new IOException(e));
+        }
+        return listing;
+    }
+
+    protected CloudStorageAccount createStorageConnection(ProcessContext context) {
```

Comment:
This could be private...  I notice that there are similar methods defined in AbstractAzureProcessor.  The connection methods could be decoupled into an interface with default methods for reuse within AbstractAzureProcessor and ListAzureBlobStorage?